app/Actions/Photo/Create.php

```diff
@@ -118,6 +121,22 @@ public function add(
 			$res = $this->save($this->photo);
 		}
 
+		//extract the main color and palette
+		$palette = ColorThief::getPalette($this->path, 5);
+
+		if (!empty($palette)) {
+			foreach ($palette as $index => $value) {
```

Comment:
Why do you need the additional `if`-clause? `foreach` alone should be fine. If `$palette` is empty, then `foreach` does not iterate.